Create convert_hipchat_data.

showell opened the pull request:
I am breaking this into several commits to make it easier to back out thing where we may have decisions, or where it might be convenient for cribbing in future conversions, but each commit, including the first commit, is atomic to the best of my knowledge.

timabbott commented on zerver/data_import/hipchat.py:
```diff
@@ -0,0 +1,236 @@
+        **in_dict
+    )
+    dct = model_to_dict(obj)
+    return dct
```
I would probably just integrate this bit into the `process` function above; it seems like the `process` function could do this formatting as well, right?

timabbott commented on zerver/data_import/hipchat.py:
```diff
@@ -0,0 +1,236 @@
+        d
+        for d in flat_data
+        if not d['is_archived']
+    ]
```
We probably want to import `is_archived` rooms as `deactivated=True` streams in Zulip.

timabbott commented on zerver/data_import/hipchat.py:
```diff
@@ -0,0 +1,236 @@
+        **in_dict
+    )
+    dct = model_to_dict(obj)
+    return dct
```
Same story as with `UserProfile`; this should probably be calling `build_stream` inside `process` above.  

(I want to use the `build_stream` type helpers from `import_util` in part for code deduplication and in part because we plan to add additional sanity-check validation to those functions that would be annoying to have in each export tool).

timabbott commented on zerver/data_import/import_util.py:
```diff
@@ -57,6 +57,37 @@ def build_recipient(type_id: int, recipient_id: int, type: int) -> ZerverFieldsT
+        recipients.append(recipient_dict)
+        recipient_id += 1
+
+    return recipients
```
I assume the longer-term plan is to reuse this function for the Slack/Gitter imports and cut a bunch of code duplication?  Seems reasonable.

showell replied:
yeah, I added a relevant comment

showell commented:
I believe I've addressed your first round of feedback, @timabbott, and I also added a commit to auto-subscribe users to public streams, which I discussed on chat with @rishig.

There's still more work to do here, but the current commits should be merge-able.

timabbott commented:
Looks reasonable, merged, thanks @showell!  

I think a good next step here is to look at whether HipChat's API has a way to get at which users are subscribed to which rooms (so that users could provide that data or the relevant API keys as a separate argument); that seems like the essential bit we need to make the import experience pretty reasonable.  (We have to do something similar with Slack import to get email addresses for users).

timabbott commented:
@rheaparekh I think you'll definitely enjoy reading this and thinking about code deduplication for it.
